operator/pkg/lbipam/lbipam.go

```diff
@@ -0,0 +1,1673 @@
+		ipNetStr(targetRange.allocRange.CIDR()),
+		ipNetStr(collisionRange.allocRange.CIDR()),
+		collisionPool.Name,
+	)
+
+	// Mark all ranges of the pool as internally disabled so we will not allocate from them.
+	targetPoolRanges, _ := ipam.RangesStore.GetRangesForPool(targetPool.UID)
+	for _, poolRange := range targetPoolRanges {
+		poolRange.internallyDisabled = true
+	}
+
+	err := ipam.patchPoolStatus(ctx, targetPool)
+	if err != nil {
+		ipam.Logger.WithError(err).Error("Error while updating IP pool status (mark conflict)")
+		return
+	}
+}
+
+// unmarkPool removes the "Conflicting" status from the pool and removes the internally disabled flag from its ranges
+func (ipam *LBIPAM) unmarkPool(ctx context.Context, targetPool *cilium_api_v2alpha1.CiliumLoadBalancerIPPool) {
+	targetPool.Status.Conflicting = false
+	targetPool.Status.ConflictReason = ""
+
+	// Re-enabled all ranges
+	targetPoolRanges, _ := ipam.RangesStore.GetRangesForPool(targetPool.UID)
+	for _, poolRange := range targetPoolRanges {
+		poolRange.internallyDisabled = false
+	}
+
+	err := ipam.patchPoolStatus(ctx, targetPool)
+	if err != nil {
+		ipam.Logger.WithError(err).Error("Error while updating IP pool status (unmark)")
+		return
+	}
+}
+
+func (ipam *LBIPAM) patchSvcStatus(ctx context.Context, sv *ServiceView) error {
+	replaceSvcStatus := []k8s.JSONPatch{
+		{
+			OP:    "replace",
+			Path:  "/status",
+			Value: sv.Status,
+		},
+	}
+
+	createStatusPatch, err := json.Marshal(replaceSvcStatus)
+	if err != nil {
+		return fmt.Errorf("json.Marshal(%v) failed: %w", replaceSvcStatus, err)
+	}
+
+	_, err = ipam.SvcClient.Services(sv.Key.Namespace).Patch(ctx, sv.Key.Name,
+		types.JSONPatchType, createStatusPatch, meta_v1.PatchOptions{
+			FieldManager: ciliumFieldManager,
+		}, "status")
+
+	return err
+}
+
+func (ipam *LBIPAM) patchPoolStatus(ctx context.Context, pool *cilium_api_v2alpha1.CiliumLoadBalancerIPPool) error {
+	replaceSvcStatus := []k8s.JSONPatch{
+		{
+			OP:    "replace",
+			Path:  "/status",
+			Value: pool.Status,
+		},
+	}
+
+	createStatusPatch, err := json.Marshal(replaceSvcStatus)
+	if err != nil {
+		return fmt.Errorf("json.Marshal(%v) failed: %w", replaceSvcStatus, err)
+	}
+
+	_, err = ipam.PoolClient.Patch(ctx, pool.Name,
+		types.JSONPatchType, createStatusPatch, meta_v1.PatchOptions{
+			FieldManager: ciliumFieldManager,
+		}, "status")
+
+	return err
+}
+
+// svcLabels clones the services labels and adds a number of internal labels which can be used to select
+// specific services and/or namespaces using the label selectors.
+func svcLabels(svc *core_v1.Service) slim_labels.Set {
+	clone := maps.Clone(svc.Labels)
+	clone[serviceNameLabel] = svc.Name
+	clone[serviceNamespaceLabel] = svc.Namespace
+	return clone
+}
+
+// PoolStore is a storage structure for IPPools
+type PoolStore struct {
+	// Map of all IP pools
+	pools map[types.UID]*cilium_api_v2alpha1.CiliumLoadBalancerIPPool
```

Comment:
As a general rule, it's not common to ever use Kubernetes UIDs. There are a number of reasons for this, but here's one example: Informers don't use them. This means that updates are coalesced, and deletes are elided.

Consider the following timeline, in which a connection to the apiserver is disrupted and the Informer does a re-List:

1. Object A has UID 1
2. Client disconnects
3. Object A is deleted
4. Object A is created. It now has UID 2
5. Client reconnects
6. Informer consumer gets an Update, not a Delete / Create.

I've seen a lot of bugs pop up when using UIDs, so unless there's a really good reason, I would suggest not.

Reply:
I have not done to much K8s agent/operator work before this. I used UIDs since explicit unique identifier tend to be better primary keys in database bases and names which derive from user input.

I am not following the exact issue you are describing, but I believe you if think it can cause issues. Refactoring the key used shouldn't be to big of a pain.